src/MappedDict.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+/**
+ * @brief A simple class to deal with 1-1 value mapping, a la python dictionaries
+ * @details Operations operate on pointers so that this class has no data storage,
+ * data should be set up by the caller and passed in to the constructor.
+ * @tparam KeyType Type that should be the 'key'
+ * @tparam ValueType Type that should be the 'value'
+ */
+template<class KeyType, class ValueType>
```

Comment:
Uh-oh, we have the first template in the project... it's all downhill from here.... :-)

Reply:
Haha yeah I thought that since this is a pretty simple and easy to follow implementation of templates it would be a good first introduction to them in this project :)